pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -48,10 +54,23 @@ func (h *AnnotationBackend) RecordStatesAsync(ctx context.Context, rule history_
 	// Build annotations before starting goroutine, to make sure all data is copied and won't mutate underneath us.
 	annotations := buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
+
 	errCh := make(chan error, 1)
 	go func() {
 		defer close(errCh)
-		errCh <- h.recordAnnotationsSync(ctx, panel, annotations, logger)
+
+		start := h.clock.Now()
+		defer func() {
+			dur := h.clock.Now().Sub(start)
+			h.metrics.PersistDuration.Observe(dur.Seconds())
+		}()
+
+		org := fmt.Sprint(rule.OrgID)
+		defer h.metrics.WritesTotal.Inc()
+		defer h.metrics.TransitionsTotal.WithLabelValues(org).Add(float64(len(annotations)))
```

Comment:
```suggestion
	    h.metrics.WritesTotal.Inc()
		h.metrics.TransitionsTotal.WithLabelValues(org).Add(float64(len(annotations)))
```

Reply:
I don't think you want to wait until the operation is complete otherwise you might end up with a failure before a total  which looks incorrect.

Reply:
I also don't love this separation - you increment failure in one place but increment total in another, this seems confusing. Would it be too hard to increment in the same place?

Reply:
Yeah, this is an artifact of when we were creating the timer (which got moved in a previous review round). In that implementation I tried to put all the metrics calculation in one place where I could.

Fixed.